src/realm/obj.cpp

```diff
@@ -1936,6 +1936,84 @@ bool Obj::remove_backlink(ColKey col_key, ObjLink old_link, CascadeState& state)
 
 void Obj::assign(const Obj& other)
 {
+    copy(other);
+    auto copy_links = [this, &other](ColKey col) {
+        auto t = m_table->get_opposite_table(col);
+        auto c = m_table->get_opposite_column(col);
+        auto backlinks = other.get_all_backlinks(col);
+        for (auto bl : backlinks) {
+            auto linking_obj = t->get_object(bl);
+            if (c.get_type() == col_type_LinkList) {
+                auto linking_obj_list = linking_obj.get_linklist(c);
+                auto n = linking_obj_list.find_first(other.get_key());
+                REALM_ASSERT(n != realm::npos);
+                linking_obj_list.set(n, get_key());
+            }
+            else if (c.get_attrs().test(col_attr_Set)) {
+                REALM_UNREACHABLE();
+            }
+            else if (c.get_attrs().test(col_attr_Dictionary)) {
+                auto linking_obj_dictionary = linking_obj.get_dictionary_ptr(c);
+                auto pos = linking_obj_dictionary->find_any(ObjLink{other.get_table()->get_key(), other.get_key()});
+                REALM_ASSERT(pos != realm::npos);
+                Mixed key = linking_obj_dictionary->get_key(pos);
+                linking_obj_dictionary->insert(key, ObjLink{m_table->get_key(), get_key()});
+            }
+            else if (c.get_type() == col_type_Mixed && linking_obj.get_any(c).get_type() == type_TypedLink) {
+                REALM_ASSERT(!linking_obj.get<ObjKey>(c) || linking_obj.get<ObjKey>(c) == other.get_key());
+                linking_obj.set_any(c, ObjLink{m_table->get_key(), get_key()});
+            }
+            else if (c.get_type() == col_type_Link) {
+                // Single link
+                REALM_ASSERT(!linking_obj.get<ObjKey>(c) || linking_obj.get<ObjKey>(c) == other.get_key());
+                linking_obj.set(c, get_key());
+            }
+        }
+        return false;
+    };
+    m_table->for_each_backlink_column(copy_links);
+}
+
+void Obj::handle_multiple_backlinks_during_schema_migration()
+{
+    REALM_ASSERT(!m_table->get_primary_key_column());
+    auto copy_links = [this](ColKey col) {
+        auto opposite_table = m_table->get_opposite_table(col);
+        auto opposite_column = m_table->get_opposite_column(col);
+        auto backlinks = get_all_backlinks(col);
+        for (auto backlink : backlinks) {
+            auto obj = m_table->create_object();
+            obj.copy(*this);
+            auto linking_obj = opposite_table->get_object(backlink);
+            fix_linking_object_during_schema_migration(linking_obj, obj, opposite_column);
+        }
+        return false;
+    };
+    m_table->for_each_backlink_column(copy_links);
+}
+
+void Obj::copy(const Obj& other)
+{
+    // To be used carefully since it could potentially duplicate user data.
```

Comment:
Let's document this new behaviour in the header file so that there are no surprises for SDKs who try to use it in the future.

Reply:
this should now be resolved.